lib/firestore/firestore_gateway.dart

```diff
@@ -150,6 +160,49 @@ class FirestoreGateway {
         .toList();
   }
 
+  Future<T> runTransaction<T>(
```

Comment:
I appreciate recursiveness as much as anyone but here it feels a bit unwarranted, especially since it exposes `attempt` and `maxAttemps` in a "public" method (the class isn't supposed to be called directly but still).

Since you have the transaction itself being run in a private method `_runTransaction`, would it make sense to wrap the retries in a loop instead of having the function calling itself?

Reply:
@jsgalarraga thank you for getting back to me, and once again sorry for allowing this to go unreviewed for so long.

I appreciate you trying out a loop-based solution but if you think that doesn't work out I'm happy accepting the patch as-is.